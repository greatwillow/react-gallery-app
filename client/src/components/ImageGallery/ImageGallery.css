.empty-content-pane {
  position: absolute;
  top: 30px;
  left: 0px;
  width: 100%;
  height: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 30px;
  color: #fff;
  font-family: 'Architects Daughter', sans-serif;
  text-align: center;
}

.empty-content-pane > h2 {
  color: rgb(17, 230, 148);
}
