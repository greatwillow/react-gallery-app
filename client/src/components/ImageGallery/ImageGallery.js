import React, { Component } from 'react';
// COMPONENTS
import ImageCardGrid from '../ImageCardGrid/ImageCardGrid';
// STYLES
import './ImageGallery.css';

class ImageGallery extends Component {
  render() {
    console.log('GALLERY WINDOW ', this.props.windowWidth, ' and props ', this.props);
    if (this.props.photos && this.props.photos.length > 0) {
      return <ImageCardGrid {...this.props} />;
    } else {
      return (
        <div className="empty-content-pane">
          <h2>Welcome to Bella Galleria</h2>
          <h3>Enjoy searching through a selection of beautiful photos!</h3>
        </div>
      );
    }
  }
}

export default ImageGallery;
