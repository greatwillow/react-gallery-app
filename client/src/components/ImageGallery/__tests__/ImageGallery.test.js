import React from 'react';
import ImageGallery from '../ImageGallery';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Image Gallery component renders correctly', () => {
  const props = {
    handleProvidePhoto: jest.fn(),
    handleShowModal: jest.fn(),
    photos: null,
    windowWidth: 800,
    errorMessage: {
      errorFound: false,
      message: 'test message'
    }
  };
  const imageGallery = shallow(<ImageGallery {...props} />);

  it('renders correctly', () => {
    expect(imageGallery).toMatchSnapshot();
  });
});
