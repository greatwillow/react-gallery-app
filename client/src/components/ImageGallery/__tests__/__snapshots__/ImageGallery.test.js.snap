// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery component renders correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageGallery
    errorMessage={
        Object {
            "errorFound": false,
            "message": "test message",
          }
    }
    handleProvidePhoto={[Function]}
    handleShowModal={[Function]}
    photos={null}
    windowWidth={800}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Welcome to Bella Galleria
</h2>,
        <p>
           A searchable photo gallery sourced from www.pexels.com.
</p>,
        <h3>
          Enjoy searching through a selection of beautiful photos!
</h3>,
      ],
      "className": "empty-content-pane",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Bella Galleria",
        },
        "ref": null,
        "rendered": "Welcome to Bella Galleria",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " A searchable photo gallery sourced from www.pexels.com.",
        },
        "ref": null,
        "rendered": " A searchable photo gallery sourced from www.pexels.com.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enjoy searching through a selection of beautiful photos!",
        },
        "ref": null,
        "rendered": "Enjoy searching through a selection of beautiful photos!",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Welcome to Bella Galleria
</h2>,
          <p>
             A searchable photo gallery sourced from www.pexels.com.
</p>,
          <h3>
            Enjoy searching through a selection of beautiful photos!
</h3>,
        ],
        "className": "empty-content-pane",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Bella Galleria",
          },
          "ref": null,
          "rendered": "Welcome to Bella Galleria",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " A searchable photo gallery sourced from www.pexels.com.",
          },
          "ref": null,
          "rendered": " A searchable photo gallery sourced from www.pexels.com.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enjoy searching through a selection of beautiful photos!",
          },
          "ref": null,
          "rendered": "Enjoy searching through a selection of beautiful photos!",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
